Document how to run integration tests locally

greg0ire opened the pull request:
I am also replacing the commented out connection parameters in
phpunit.xml.dist with a link to this paragraph because they have
diverged from the one we use in the CI since #4067.

greg0ire committed:
Document how to run integration tests locally

I am also replacing the commented out connection parameters in
phpunit.xml.dist with a link to this paragraph because they have
diverged from the one we use in the CI since f88dc282080826528fe7326cb9a7c465aa3074d7.

greg0ire committed:
Remove unused script

I forgot to remove it in e7aef4e8ba64ff5ba72ea427cfa2a09b42801522.

greg0ire commented on phpunit.xml.dist:
```diff
@@ -22,24 +22,11 @@
-        <var name="tmpdb_port" value="3306"/>
-        <var name="tmpdb_user" value="root" />
-        <var name="tmpdb_password" value="" />
-        <var name="tmpdb_dbname" value="doctrine_tests_tmp" />
```
Another reason for this removal is a discussion with  @alli83 about how to run the test suite, where I recommended either using a CI file or these instructions. She reported that the latter did not work (and the former works :slightly_smiling_face: ).

derrabus reviewed (dismissed):
> We do not currently have a recommendation on how to spin up a database
server.

We could recommend Docker though.

greg0ire commented:
@derrabus if I'm honest, using Docker is exactly how I would do it, so I think it's fair to recommend it indeed. I also removed a previous sentence that felt out of place and replaced it with another one about versions.

derrabus commented on docs/en/reference/testing.rst:
```diff
@@ -62,6 +62,33 @@ multiple concurrent database connections, transactions, locking, performance-rel
+
+We do not currently have specific instructions on how to run a Database
+server, but we do recommend Docker as a convenient way to do so.
+To know which version of a database you should use, you can check
+``.github/workflows/continuous-integration.yml`` to see which versions
+we officially support. For instance, the
+supported PostgreSQL versions are listed under
+``jobs.phpunit-postgres.strategy.matrix.postgres-version``.
```
_Supported_ versions are listed here: https://www.doctrine-project.org/projects/doctrine-dbal/en/4.0/reference/platforms.html

The CI config tells you what versions we _test against_.

greg0ire replied:
Addressed.

derrabus commented on docs/en/reference/testing.rst:
```diff
@@ -62,6 +62,31 @@ multiple concurrent database connections, transactions, locking, performance-rel
+server, but we do recommend Docker as a convenient way to do so.
+We do not recommend running against a particular version of the chosen
+RDBMS either, as long as you pick one of the `officially supported versions
+<https://www.doctrine-project.org/projects/doctrine-dbal/en/4.0/reference/platforms.html>`_.
```
Shouldn't this be an internal link that should be resolved during rendering?

greg0ire replied:
Oh right, definitely!
